ingestion/src/metadata/interfaces/datalake/datalake_profiler_interface.py

```diff
@@ -0,0 +1,198 @@
+        logger.debug(f"Running profiler for {table}")
+        try:
+
+            row = compute_metrics_registry.registry[metric_type.value](
+                metrics,
+                session=self.client,
+                data_frame_list=self.data_frame_list,
+                column=column,
+                processor_status=self.processor_status,
+            )
+        except Exception as err:
+            logger.error(err)
+            row = None
+        if column:
+            column = column.name
+        return row, column
+
+    def fetch_sample_data(self, table) -> TableData:
+        """Fetch sample data from database
+
+        Args:
+            table: ORM declarative table
+
+        Returns:
+            TableData: sample table data
+        """
+        sampler = Sampler(
+            session=self.client,
+            table=self.data_frame_list,
+            profile_sample=self.profile_sample,
+            partition_details=self.partition_details,
+            profile_sample_query=self.profile_query,
+        )
+        return sampler.fetch_dl_sample_data()
+
+    def get_composed_metrics(
+        self, column: Column, metric: Metrics, column_results: Dict
+    ):
+        """Given a list of metrics, compute the given results
+        and returns the values
+
+        Args:
+            column: the column to compute the metrics against
+            metrics: list of metrics to compute
+        Returns:
+            dictionary of results
+        """
+        try:
+            return metric(column).fn(column_results)
+        except Exception as exc:
+            logger.debug(traceback.format_exc())
+            logger.warning(f"Unexpected exception computing metrics: {exc}")
+            return None
+
+    def get_all_metrics(
+        self,
+        metric_funcs: list,
+    ):
+        """get all profiler metrics"""
+        self.fetch_dl_data_frame(self.service_connection_config.configSource)
+        profile_results = {"table": {}, "columns": defaultdict(dict)}
+        metric_list = [
+            self.compute_metrics(metric_funcs=metric_func)
+            for metric_func in metric_funcs
+        ]
+        for metric_result in metric_list:
+            profile, column = metric_result
+
+            if not column:
+                profile_results["table"].update(profile)
+            else:
+                if profile:
+                    profile_results["columns"][column].update(
+                        {
+                            "name": column,
+                            "timestamp": datetime.now(tz=timezone.utc).timestamp(),
+                            **profile,
+                        }
+                    )
+        return profile_results
+
+    @property
+    def table(self):
+        """OM Table entity"""
+        return self._table
+
+    def get_columns(self):
+        return [
+            ColumnBaseModel(name=column.name.__root__, datatype=column.dataType)
+            for column in self._table.columns
```

Comment:
here you are getting the columns for OM Table Entity. I think instead you need to get it from the data frame as computation will be performed on the data frame (not the OM Table Entity)

Reply:
@TeddyCr get_columns is just for the checking if the data type is quanitfiable or concatenable, the computation is directly happening from dataframe